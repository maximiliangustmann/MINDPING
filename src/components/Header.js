import React from 'react'
import { NavLink } from 'react-router-dom'
import styled from 'styled-components/macro'
import { ReactComponent as Home } from '../img/home.svg'
import { ReactComponent as Add } from '../img/add.svg'
import SortButton from './SortButton'
import PropTypes from 'prop-types'

Header.propTypes = {
  sortBySubcategory: PropTypes.func,
  sortByRating: PropTypes.func,
}

export default function Header({ sortBySubcategory, sortByRating }) {
  return (
    <HeaderWrapper>
      <NavLink exact to="/" data-cy="homebutton">
        <HomeIcon />
      </NavLink>
      <NavLink to="/create" data-cy="addbutton">
        <AddIcon data-cy="add-icon" />
      </NavLink>
      <SortButton
        sortBySubcategory={sortBySubcategory}
        sortByRating={sortByRating}
      />
    </HeaderWrapper>
  )
}

const HomeIcon = styled(Home)`
  fill: var(--main-red);
  height: 40px;
  width: 40px;
  margin: 20px;
`

const AddIcon = styled(Add)`
  fill: var(--main-red);
  height: 40px;
  width: 40px;
  margin: 20px;
`

const HeaderWrapper = styled.div`
  display: flex;
  justify-content: center;
  margin-top: 20px;
`
