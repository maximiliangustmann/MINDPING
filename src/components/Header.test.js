import React from 'react'
import renderer from 'react-test-renderer'
import Header from './Header'
import { MemoryRouter } from 'react-router-dom'

describe('Header', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <MemoryRouter>
        <Header />
      </MemoryRouter>
    )
    expect(tree).toMatchSnapshot()
  })
})
