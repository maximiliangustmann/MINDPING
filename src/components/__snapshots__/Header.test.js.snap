// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="Header__HeaderWrapper-sc-1uw4t5o-2 bYSPhJ"
>
  <a
    aria-current="page"
    className="active"
    data-cy="homebutton"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <svg
      className="Header__HomeIcon-sc-1uw4t5o-0 jgwhff"
    >
      home.svg
    </svg>
  </a>
  <a
    aria-current={null}
    data-cy="addbutton"
    href="/create"
    onClick={[Function]}
  >
    <svg
      className="Header__AddIcon-sc-1uw4t5o-1 RFjBk"
      data-cy="add-icon"
    >
      add.svg
    </svg>
  </a>
  <div
    className="sc-AxirZ hlnYRa"
  >
    <svg
      className="sc-AxjAm fZvzJc"
      data-cy="sort-icon"
      onClick={[Function]}
    >
      sort.svg
    </svg>
  </div>
</div>
`;
